### PR TITLE
GHA workflow improvements

### PR DESCRIPTION
**Pull request by meiao**
### Overview
- Simplifies the AIT workflow to use standard environment variables, instead of creating its own.
- Fails AITs and verify instrumentation tests when the agent jar cache misses.

### CODE REVIEW

File: .github/workflows/Java-Instrumentation-Tests.yml [hunk 1]
```diff
@@ -50,6 +50,9 @@ jobs:
       - name: Build newrelicJar
         run: ./gradlew $GRADLE_OPTIONS clean jar
 
+      - name: Remove parallel from GRADLE_OPTIONS
+        run: echo "GRADLE_OPTIONS=$GRADLE_OPTIONS" | sed "s/--parallel//" >> $GITHUB_ENV
+
```
Comment (meiao) [inline .github/workflows/Java-Instrumentation-Tests.yml:55]: This prevents the integration tests from running in parallel, thus preventing any 2 of  them from using the same resource at the same time.

File: .github/workflows/Test-AITs.yml [hunk 2]
```diff
@@ -109,6 +110,14 @@ jobs:
...
       matrix: ${{ fromJson(needs.list-tests.outputs.tests) }}
     steps:
+      - name: Retrieve agent from cache
+        id: retrieve-agent
+        uses: actions/cache@v3
+        with:
+          path: /home/runner/work/newrelic-java-agent/newrelic-java-agent/newrelic-agent/build/newrelicJar/newrelic.jar
+          key: ${{ github.run_id }}
+          fail-on-cache-miss: true
+
```
Comment (meiao) [inline .github/workflows/Test-AITs.yml:120]: Fail fast if cache misbehaves.

File: .github/workflows/Test-AITs.yml [hunk 3]
```diff
@@ -192,40 +187,15 @@ jobs:
...
-          java-version: 17
-
-      # Set the JDK variable
-      - name: Set the required JDK variable
-        run: |
-          echo "ORG_GRADLE_PROJECT_jdk17=${JAVA_HOME}" >> $GITHUB_ENV
-
-      # Install latest non-LTS Java version (we should only ever test one non-LTS version to keep test times reasonable)
-      - name: Set up Java 20
+      - name: Set up Javas
```
Comment (meiao) [inline .github/workflows/Test-AITs.yml:190]: Simplifying Java setup.

File: .github/workflows/X-Reusable-VerifyInstrumentation.yml [hunk 4]
```diff
@@ -68,6 +68,7 @@ jobs:
         with:
           path: /home/runner/work/newrelic-java-agent/newrelic-java-agent/newrelic-agent/build/newrelicJar/newrelic.jar
           key: ${{ github.run_id }}
+          fail-on-cache-miss: true
```
Comment (meiao) [inline .github/workflows/X-Reusable-VerifyInstrumentation.yml:71]: fail fast when agent cache misses.

**Review by jasonjkeller — APPROVED** [review]

**Review by jtduffy — APPROVED** [review]

**Pull request merged**